worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -697,6 +697,10 @@ private[deploy] class Controller(
       val releaseReplicaLocations =
         partitionLocationInfo.removeReplicaPartitions(shuffleKey, replicaLocations)
       workerInfo.releaseSlots(shuffleKey, releaseReplicaLocations._1)
+      workerSource.incCounter(
+        WorkerSource.SLOTS_ALLOCATED,
+        -(primaryLocations.size() + replicaLocations.size() - failedPrimaries.size()
```

Comment:
I'm not sure if we should decrease `SLOTS_ALLOCATED`. Currently this metric only increases. If we decrease here, should we also decrease everywhere `releaseSlots` is called? cc @pan3793 @cfmcgrady @AngersZhuuuu